File: benchmark/swift/task6_fileio.swift
import Foundation

func countWords(inputFile: String, outputFile: String) throws {
    let text = try String(contentsOfFile: inputFile)
    let words = text.split(separator: " ").map(String.init)
    var counts: [String: Int] = [:]

    for word in words {
        counts[word, default: 0] += 1
    }

    let output = counts.map { "\($0.key): \($0.value)" }.joined(separator: "\n")
    try output.write(toFile: outputFile, atomically: true, encoding: .utf8)
}

File: benchmark/php/task6_fileio.php
<?php
function count_words(string $inputFile, string $outputFile): void {
    $text = file_get_contents($inputFile);
    $words = preg_split('/\s+/', $text, -1, PREG_SPLIT_NO_EMPTY);
    $counts = array_count_values($words);

    $output = '';
    foreach ($counts as $word => $count) {
        $output .= "$word: $count\n";
    }
    file_put_contents($outputFile, $output);
}
